tests/php/src/Admin/ReaderThemesTest.php

```diff
@@ -104,6 +107,90 @@ public function test_get_themes() {
 		$this->assertNotContains( 'with-legacy', $available_theme_slugs );
 	}
 
+	/**
+	 * Test that themes API success does not result in a WP_Error.
+	 *
+	 * @covers ::get_themes
+	 * @covers ::get_default_reader_themes
+	 */
+	public function test_themes_api_success() {
+		$this->reader_themes->get_themes();
+
+		$this->assertNull( $this->reader_themes->get_themes_api_error() );
+	}
+
+	/**
+	 * Test that a themes API failure results in a WP_Error.
+	 *
+	 * @covers ::get_themes
+	 * @covers ::get_default_reader_themes
+	 */
+	public function test_themes_api_failure() {
+		add_filter( 'themes_api_result', '__return_null' );
+
+		$this->reader_themes->get_themes();
+
+		$error = $this->reader_themes->get_themes_api_error();
+		$this->assertWPError( $error );
+		$this->assertEquals(
+			'The request for reader themes from the WordPress.org resulted in an invalid response. Check your Site Health to confirm that your site can communicate with WordPress.org. Otherwise, please try again later or contact your host.',
+			$error->get_error_message()
+		);
+
+		remove_filter( 'themes_api_result', '__return_null' );
+	}
+
+	/**
+	 * Test that a themes API response with an empty themes array results in a WP_Error.
+	 *
+	 * @covers ::get_themes
+	 * @covers ::get_default_reader_themes
+	 */
+	public function test_themes_api_empty_array() {
+		$filter_cb = static function() {
+			return (object) [ 'themes' => [] ];
+		};
+		add_filter( 'themes_api_result', $filter_cb );
+
+		$this->reader_themes->get_themes();
+
+		$error = $this->reader_themes->get_themes_api_error();
+		$this->assertWPError( $this->reader_themes->get_themes_api_error() );
+		$this->assertEquals(
+			'The default reader themes cannot be displayed because a plugin appears to be overriding the themes response from WordPress.org.',
+			$error->get_error_message()
+		);
+	}
+
+	/**
+	 * Test that an error is stored in state when themes_api returns an error.
+	 *
+	 * @covers ::get_themes
+	 * @covers ::get_default_reader_themes
+	 */
+	public function test_themes_api_wp_error() {
+		$filter_cb = static function() {
+			return new WP_Error(
+				'amp_test_error',
+				'Test message'
+			);
+		};
+		add_filter( 'themes_api_result', $filter_cb );
+
+		$this->reader_themes->get_themes();
+
+		$error = $this->reader_themes->get_themes_api_error();
+		$this->assertWPError( $this->reader_themes->get_themes_api_error() );
+		$this->assertStringStartsWith(
+			'The request for reader themes from the WordPress.org resulted in an invalid response. Check your Site Health to confirm that your site can communicate with WordPress.org. Otherwise, please try again later or contact your host.',
```

Comment:
```suggestion
			'The request for reader themes from WordPress.org resulted in an invalid response. Check your Site Health to confirm that your site can communicate with WordPress.org. Otherwise, please try again later or contact your host.',
```